File: 백준/Silver/11659. 구간 합 구하기 4/구간 합 구하기 4.cc
#include <iostream>
#include <vector>

using namespace std;

int main() {

	int n, m;
	scanf("%d %d", &n, &m);

	vector<int> num(n + 1, 0);
	for (int i = 1; i <= n; i++) {
		int input = 0;
		scanf("%d", &input);
		num[i] = num[i-1] + input;
	}

	vector<int> result(m, 0);
	for (int i = 0; i < m; i++) {
		int start = 0, end = 0;
		scanf("%d %d", &start, &end);
		result[i] = num[end] - num[start - 1];
	}

	for (const auto& it : result) {
		printf("%d\n", it);
	}


	return 0;
}

File: 백준/Silver/11659. 구간 합 구하기 4/구간 합 구하기 4.py
import sys

N, M = map(int, input().split())
arr = list(map(int, input().split()))
arr2 = [list(map(int, input().split())) for _ in range(M)] 

#sums = list(0 for _ in range(N+1))
sums = [0] + arr

for i in range(1, N+1):
    sums[i] += sums[i - 1]

result = list()
for gugan in arr2:
    start, end = gugan
    result.append(sums[end] - sums[start-1])
    
for n in result:
    print(n)

